addons/crm_iap_lead_enrich/models/crm_iap_lead_enrich.py

```diff
@@ -0,0 +1,124 @@
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
+from odoo.addons.iap import jsonrpc,InsufficientCreditError
+
+DEFAULT_ENDPOINT = 'https://iap-services.odoo.com'
+
+class Lead(models.Model):
+    _inherit = 'crm.lead'
+
+    enriched_lead = fields.Boolean(string='Lead enriched based on email', default=False)
+
+    @api.multi
+    def lead_enrich_mail(self):
+        request = {}
+        for record in self:
+            if record.probability == 0:
+                continue
+            email_domain = record.partner_address_email or record.email_from
+            if email_domain:
+                request[record.id] = email_domain.split('@')[1]
+            else:
+                record.message_post_with_view('crm_iap_lead_enrich.lead_enrich_message_no_mail', subtype_id=self.env.ref('mail.mt_note').id)        
+        try:
+            response_clearbit = record._make_request(request)
+            self._enrich_leads_from_response(response_clearbit)
+        except InsufficientCreditError:
+            for lead in self:
```

Comment:
Be consistent, either you use `for record in self` or `for lead in self`, but keep the same name (line 15 and here)